app/web_framework/cluster_master_application.py

```diff
@@ -418,3 +421,10 @@ def post(self):
             [int(slave_id) for slave_id in self.decoded_body.get('slaves')]
 
         self._cluster_master.set_shutdown_mode_on_slaves(slaves_to_shutdown)
+
+
+class _SlavesHeartbeatHandler(_ClusterMasterBaseAPIHandler):
+    @authenticated
+    def post(self, slave_id):
+        slave = self._cluster_master.get_slave(int(slave_id))
+        slave.set_last_heartbeat_time()
```

Comment:
Also, you may want to add some unit tests for the Slave class. I think I saw you mocked out everything on the Slave instance so that logic is not tested via your current tests?

Reply:
um.. yeah.. the reason I removed the tests from the Slave class was because I moved all the logic out of there. The only methods left in there are a getter and a setter. Not sure if they really need unit tests.

Reply:
@mayurva Sure, at some point it starts to get a little into testing philosophy. I usually find myself trying to ignore the complexity of the actual implementation and write tests for class behavior. I can see an argument for not wanting to test single line methods (on the other hand those are some of the easiest tests to write!)

My personal view is that even single lines can have bugs in them or someone can refactor them incorrectly. If someone comes along to one of my single-line methods and adds an if statement, all of a sudden they now have two tests to write -- one for their new functionality and one for the original functionality. Basically my view is that if the application depends on the code then they should be covered by *some* test -- whether it's a unit/integration/functional test.

Anyway, just my two cents. Don't consider my comments blocking if you feel strongly a different way. Happy to chat more in person if you'd rather.